### PR TITLE
[WIP] Update version and fix build on osx

### PR DESCRIPTION
**Pull request by raivivek**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by raivivek** [comment]
@hmaarrfk Would you happen to know why we're getting this error:
```
/usr/bin/install: cannot create regular file `/usr/local/lib/libwx_baseu-3.1.so': Permission denied
```

**Comment by gwaygenomics** [comment]
@raivivek - great you were able to track this down. So far down the conda rabbit hole all the way from CellProfiler/CellProfiler#3730. Definitely an interesting case study.

A couple things my eyes noticed:

* It doesn't look like this PR is even kicking off an OSX build
* The "bad attempt" by @hmaarrfk in #2 doesn't have the same `/usr/bin/install: cannot create regular file `/usr/local/lib/libwx_baseu-3.1.so': Permission denied` error that you note above, and the PR passes the linux and windows builds
* It looks like @hmaarrfk has an idea of what needs to change. @hmaarrfk mentions `opengl` in https://github.com/conda-forge/wxwidgets-feedstock/issues/1#issuecomment-648849422

## Proposal

We are motivated to get CellProfiler4 up on bioconda, and it seems (🤞 ) that this is the bottom of the barrel. Maybe one of the three bullets above inspires some fix? If not, I am hoping that @mingwandroid might know exactly what is going on. In all conda recipes I've contributed to in the past, it always seems that @mingwandroid knows what to do in these tough cases. Sorry to bug with a direct tag, and thanks in advance!

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
@raivivek the reason you are getting this error is because it is trying to install the results in a system directory instead of the conda prefix.

Often there are ways to change where the results are installed with `INSTALL_PREFIX` or similar variables.

File: recipe/build.sh [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
-# export LDFLAGS="${LDFLAGS:-} -fuse-ld=gold -Wl,--as-needed"
+echo $target_platform
+
+if [[ "$target_platform" == "osx"* ]]; then
+  ./configure \
+      --prefix=${PREFIX} \
+      --with-opengl \
+      --with-osx_cocoa
+else
+  ./configure \
```
Comment (hmaarrfk) [inline recipe/build.sh:9]:
```suggestion
  ./configure \
     --prefix=${PREFIX} \ 
```

File: recipe/build.sh [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
-./configure \
-  --prefix=${PREFIX} \
-  --with-gtk="3" \
-  --with-opengl
 
 [[ "$target_platform" == "win-64" ]] && patch_libtool
 
 make -j ${CPU_COUNT}
-make install
+make install
```
Comment (hmaarrfk) [inline recipe/build.sh:18]:
```suggestion
make install
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -112,4 +96,4 @@ about:
 
 extra:
   recipe-maintainers:
-    - hmaarrfk
+    - hmaarrfk
```
Comment (hmaarrfk) [inline recipe/meta.yaml:99]:
```suggestion
    - hmaarrfk
```

**Review by hmaarrfk — COMMENTED** [review]
A few small things were missing.

File: recipe/docs/licence.txt [hunk 4]
```diff
@@ -0,0 +1,53 @@
+                wxWindows Library Licence, Version 3.1
```
Comment (hmaarrfk) [inline recipe/docs/licence.txt:1]: why did these files get added? Typically, things come out of source.

**Comment by hmaarrfk** [comment]
I'm not sure why osx isn't using the version you specified.

**Comment by raivivek** [comment]
Thanks for pushing on this work as well. Do you think directly specifying `-mmacosx-version-min=10.11` in the build flags should help?

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
> -mmacosx-version-min=10.11 in the build flags should help

You already specified 10.11 in the conda files, and 10.11 > 10.10, so I'm not sure why it is failing honestly.

Some funky stuff goes on in the OSX builds.

**Comment by hmaarrfk** [comment]
Do you have documentation that states that the OSX min version should be 10.11? if so, it seems you should export the variable

**Comment by hmaarrfk** [comment]
```
wxUSE_MACOSX_VERSION_MIN=10.11
```

**Comment by hmaarrfk** [comment]
@raivivek I would suggest decoupling the version update from this merge request from OSX.

Do you want a version update, or OSX? which is more important?

Lets do that first.

In #2, I managed to get OSX working rather smoothly.

The only issue is that it seems to create

```
3.1.3.0.0.dylib
```
and i think we would prefer files to have `3.1.dylib` so that they can be smoothly updated through differetn versions of wxWidgets. Would you be able to look into that?

**Comment by raivivek** [comment]
@hmaarrfk > Do you want a version update, or OSX? which is more important?
Unfortunately, that's exactly my use case right now — latest version of the package on macOS. However, it's not blocking me since I've found other workarounds. You can then focus on fixing this issue however you prefer.

> Would you be able to look into that?
I will try to look into this. It might be a while though..

**Comment by hmaarrfk** [comment]
> However, it's not blocking me since I've found other workarounds.

Which means you can use the old version on OSX? Wouldn't it be beneficial for you to help build the old version.

Unfortunately, i don't use OSX at all, and my knowledge is very skim. I probably won't be spending too much time on it. Help is really appreciated.

I'm tempted to merge #2, even though osx isn't built.

**Comment by raivivek** [comment]
My specific need was related to the wxPython package (which needs to be updated to v4.1.0, see [discussion](https://github.com/conda-forge/wxpython-feedstock/pull/48)), and I thought that maybe fixing this will give me some idea of how to update that conda recipe. My workaround around that, unfortunately, is to just create a Docker container where I directly install the pre-built wxPython wheels available from their website and avoid building from source.

I think merging #2 is fine. We can continue to look into the macOS build failures on the side with whatever time we can spare. I appreciate your help so far regarding this.

**Pull request closed without merging**